Projects/ChangeTier/Sources/Navigation/ChangeTierNavigation.swift

```diff
@@ -41,13 +42,24 @@ public class ChangeTierNavigationViewModel: ObservableObject {
             self.changeTierContractsInput = changeTierContractsInput
         }
         router = Router()
+        self.useOwnNavigation = true
     }
 
     public static func getTiers(input: ChangeTierInputData) async throws -> ChangeTierIntentModel {
         let client: ChangeTierClient = Dependencies.shared.resolve()
         let data = try await client.getTier(input: input)
         return data
     }
+
+    func missingQuotesGoBackPressed() {
```

Comment:
Is this not the same as?

```
    func missingQuotesGoBackPressed() {
        if useOwnNavigation {
            router.dismiss()
        } else {
            router.pop()
        }
    }
```

I think that would be cleaner and I don't see the purpose of checking `changeTierContractsInput `

Reply:
Not really since we could use own navigation and still have changeTierContractsInput

Reply:
But yeah, can be merged 👍